src/gui/updater/ocupdater.cpp

```diff
@@ -192,8 +192,28 @@ void OCUpdater::slotStartInstaller()
     settings.setValue(autoUpdateAttemptedC, true);
     settings.sync();
     qCInfo(lcUpdater) << "Running updater" << updateFile;
-    QProcess::startDetached(updateFile, QStringList() << "/S"
-                                                      << "/launch");
+
+    if(updateFile.endsWith(".exe")) {
+        QProcess::startDetached(updateFile, QStringList() << "/S"
+                                                          << "/launch");
+    } else if(updateFile.endsWith(".msi")) {
+        // When MSIs are installed without gui they cannot launch applications
+        // as they lack the user context. That is why we need to run the client
+        // manually here. We wrap the msiexec and client invocation in a powershell
+        // script because owncloud.exe will be shut down for installation.
+        // | Out-Null forces powershell to wait for msiexec to finish.
+        auto preparePathForPowershell = [](QString path) {
+            path.replace("'", "''");
+
+            return QDir::toNativeSeparators(path);
+        };
+
+        QString command = QString("&{msiexec /norestart /passive /i '%1'| Out-Null ; &'%2'}")
+            .arg(preparePathForPowershell(updateFile))
+            .arg(preparePathForPowershell(QCoreApplication::applicationFilePath()));
+
+        QProcess::startDetached("powershell.exe", QStringList{"-Command", command});
+    }
```

Comment:
and else?

Reply:
I could just open the file with `openURL` but not sure if we really want that. 
Or I could show a QMessageBox ... but actually this should never happen, because there's no really reasonable way to handle *anything*